packages/webdriverio/src/scripts/resq.js

```diff
@@ -3,35 +3,25 @@ export const waitToLoadReact = function waitToLoadReact () {
 }
 
 export const react$ = function react$ (selector, props, state) {
-    let elems = window.resq.resq$$(selector)
+    let element = window.resq.resq$(selector)
         .byProps(props)
         .byState(state)
 
-    if (elems.length === 0) {
+    if (!element.name) {
         return { message: `React element with selector "${selector}" wasn't found` }
     }
 
-    let elem = elems[0]
-
-    if (elem.node) {
-        return elem.node
-    }
-
-    elem = elem.node || elem.children[0].node
-    return elem
+    return element.node || element.children[0].node
 }
 
 export const react$$ = function react$$ (selector, props, state) {
-    let elems = window.resq.resq$$(selector)
+    let elements = window.resq.resq$$(selector)
         .byProps(props)
         .byState(state)
 
-    return [...elems].map((elem) => {
-        if (elem.node) {
-            return elem.node
-        }
+    if (!elements.length) {
+        return { message: `React elements with selector "${selector}" wasn't found` }
```

Comment:
for fetching elements we should comply to the webdriver spec which is just return an empty array. Only the `fineElement` command would return an error if it can not be found.

Reply:
Return an empty array in both functions?

Reply:
No, just for react$$

Reply:
4fb8094